ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingEvaluator.java

```diff
@@ -94,19 +94,20 @@ protected TableFileScanHelper.PartitionFilter getPartitionFilter() {
 
   private void initPartitionPlans(TableFileScanHelper tableFileScanHelper) {
     PartitionSpec partitionSpec = arcticTable.spec();
-    for (TableFileScanHelper.FileScanResult fileScanResult : tableFileScanHelper.scan()) {
-      StructLike partition = fileScanResult.file().partition();
-      String partitionPath = partitionSpec.partitionToPath(partition);
-      PartitionEvaluator evaluator = partitionPlanMap.computeIfAbsent(partitionPath, this::buildEvaluator);
-      evaluator.addFile(fileScanResult.file(), fileScanResult.deleteFiles());
-    }
+    // add partition properties first, then add files, because files may use these properties
```

Comment:
The evaluator is generated in the following logic,
```
partitionPlanMap.computeIfAbsent(partitionPath, this::buildEvaluator);
```
so the evaluator is always null here, which will not make the properties take effect ?

Reply:
Yes, you are right.👍
My negligence, I will fix it immediately.

Reply:
I moved adding partition properties to the constructor, to avoid adding files before adding properties.